WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,132 @@
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+    <% if @competitions.length == 0 %>
+      <%= render 'index_form' %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <table class="table competitions-table table-striped table-hover table-condensed floatThead">
+          <thead>
+            <tr>
+              <th colspan="7">
+                <%= render 'index_form' %>
+              </th>
+            </tr>
+            <tr>
+              <th>Year</th>
+              <th>Date</th>
+              <th>Name</th>
+              <th></th>
+              <th>Country, City</th>
+              <th>Venue</th>
+              <th class="big-column"></th>
+            </tr>
+          </thead>
+          <tbody>
+            <%= render 'index_table', competitions: @not_past_competitions %>
+            <tr id="today-line">
+              <td class="text-center"><a id="today"></a>Today</td>
+              <td colspan="6" ></td>
+            </tr>
+            <%= render 'index_table', competitions: @past_competitions %>
+          </tbody>
+        </table>
+      <% else %>
+        <%= render 'index_form' %>
+        <div id="map">
+        </div>
+      <% end %>
     <% end %>
-  </table>
+  <% end %>
 </div>
+
+<script>
+$(function() {
+  var $today = document.getElementById("today");
+  if ($today) {
+    var elOffset = $("#today").offset().top;
+    var windowHeight = $(window).height();
+    var offset = elOffset - Math.max((windowHeight - $("#today").height()) / 2, 0);
+
+    $('html, body').animate({ scrollTop: offset}, 50);
+  }
+
+  function initMap() {
+    var $map = new google.maps.Map(document.getElementById('map'), {
+      zoom: 2,
+      center: {lat: 0, lng: 0},
+      scrollwheel: true
+    });
+
+    wca.allCompetitions(<%= @competitions.map do |c|
+    {
+      id: c.id,
+      name: c.name,
+      latitude_degrees: c.latitude_degrees,
+      longitude_degrees: c.longitude_degrees,
+      year: c.year,
+      month: c.month,
+      day: c.day,
+      cityName: c.cityName,
+      marker_date: c.start_date.to_formatted_s(:long)
+    }
+      end.to_json.html_safe %>, $map);
+  }
+
+  wca.allCompetitions = function(competitions, map) {
+    var desiredCompetitionById = _.indexBy(competitions, 'id');
+
+    var desiredIds = Object.keys(desiredCompetitionById);
+
+    var markers = [];
+
+    desiredIds.forEach(function(id) {
+      var c = desiredCompetitionById[id];
+
+      var compDate = new Date(c.year, c.month-1, c.day);
+
+      var contentString = "<a href='/competitions/" + c.id + "'>" + c.name + "</a><br />" + c.marker_date + " - " + c.cityName;
```

Comment:
nitpick: it would be nice if we used ruby to generate the url instead. We could do that earlier when you're building the JSON from all the competitions...